### ISSUE TITLE
A user should be able to tell their CSV files apart

### ISSUE THREAD

**Issue by jackfirth** [grokkerlab/grokker#59]
All we show is a name at the moment, which can be tricky if all your files are named "dataset1", "dataset2", etc.

Possible ideas for other things to show about each file:

- Date uploaded
- Number of rows
- Column names
- Column types

**Comment by TGOlson** [comment]
:+1: :+1: great idea - does this mean we need to parse some metadata out of the uploads before storing them? An alternative would be to give them a preview and make them annotate their rows with types before finalizing the upload.

**Comment by jackfirth** [comment]
Yeah, though the extraction of the metadata would probably be easier to do server side. Perhaps a sqlite database for storing metadata?

As for row types, maybe we give the user the ability to specify some types and then we check that the column is valid for that type. Then we could add on the ability to infer types and use those as defaults. Types would be useful because we could use the information to store the data much more compactly than in a csv file.
